docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,382 @@
 
-Example:
+Signature:
 
 ```fsharp
-open System.Threading
+computation: Async<'T> -> Async<Choice<'T, exn>>
+```
 
-// Create a workflow which will loop forever.
-let workflow =
+When to use:
+
+- When you are performing asynchronous work that may fail with an exception and you want to handle that exception in the caller.
+
+What to watch out for:
+
+- When using combined or sequenced asynchronous computations, the encompassing computation will fully stop if one of its "internal" computations throws an exception.
+
+### Async.Ignore
+
+Creates an asynchronous computation that runs the given computation and ignores its result.
+
+Signature:
+
+```fsharp
+computation: Async<'T> -> Async<unit>
+```
+
+When to use:
+
+- When you have an asynchronous computation whose result can be discarded.
+- In conjunction with `Async.Start` or other functions that accept an `Async<unit>`.
+
+What to watch out for:
+
+- If you must use this because you wish to use `Async.Start` or another function that requires `Async<unit>`, consider if discarding the result is okay to do. Discarding results just to fit a type signature should not generally be done.
+
+### Async.RunSynchronously
+
+Runs an asynchronous computation and awaits its result on the calling thread. This call is blocking.
+
+Signature:
+
+```fsharp
+computation: Async<'T> * timeout: ?int * cancellationToken: ?CancellationToken -> 'T
+```
+
+When to use it:
+
+- If you need it, use it only once in an application - at the entry point for an executable.
+- When you don't care about performance and want to execute a set of other asynchronous operations at once.
+
+What to watch out for:
+
+- This function blocks the calling thread, which nullifies the point of asynchrony if it is used to only execute a single `Async<'T>`.
+
+### Async.Start
+
+Starts an asynchronous computation with no return type in the thread pool. Doesn't wait for its result. Nested computations ran with `Async.Start` are ran completely independently of the parent computation that called them. If the parent computation is canceled, the nested computations aren't canceled.
+
+Signature:
+
+```fsharp
+computation: Async<unit> * cancellationToken: ?CancellationToken -> unit
+```
+
+Use only when:
+
+- You have an asynchronous computation but don't care about the result.
+- You don't care about when an asynchronous computation runs.
+- You don't care which thread an asynchronous computation runs on.
+- You don't have any need to be aware of or report exceptions resulting from the task.
+
+What to watch out for:
+
+- Exceptions raised by computations started with `Async.Start` aren't propagated to the caller.
```

Comment:
They will unwind the stack completely resulting in the process terminating.

Reply:
Unless there's a global exception handler installed in full framework. I believe classic fsi does this.